ide/projectui/src/org/netbeans/modules/project/ui/OpenProjectList.java

```diff
@@ -1379,6 +1376,7 @@ private boolean doOpenProject(final @NonNull Project p) {
             public @Override Boolean run() {
             log(Level.FINER, "already opened: {0} ", openProjects);
             for (Project existing : openProjects) {
+                // TODO double tap ???
                 if (p.equals(existing) || existing.equals(p)) {
```

Comment:
this is either a harmless mistake or a red flag

Reply:
I often turn to @emilianbold's helpful [git dump](https://github.com/emilianbold/netbeans-releases) of the old NetBeans Mercurial logs for curiosities like these. Here's the relevant commit:

https://github.com/emilianbold/netbeans-releases/commit/56b427cd661de066f5e5ec644099f74a37000605

There's a Bugzilla issue number 156536 referenced, but the [archive](https://bz.apache.org/netbeans/describecomponents.cgi) doesn't seem to work anymore.

Reply:
ok. so its a red flag of a broken equals contract too. Not only hash code.